static/components/TaskDetails.jsx

```diff
@@ -127,37 +137,113 @@ var TaskDetails = React.createClass({
   updateBuildLog(data) {
     // add each additional new line
     var frag       = document.createDocumentFragment();
-    var text       = data.chunks
-    var objLength  = data.chunks.length
+    var text       = data.chunks;
+    var objLength  = data.chunks.length;
+    var id         = 0;
+    var timeid     = 0;
+    var index      = 0;
 
     for(var i = 0; i < objLength; i++){
-      var timer    = new Date(data.chunks[i].date)
-      var timeMil  = timer.getTime()
+      if(data.chunks[i] != undefined){
+        this.setState({
+          lines: [...this.state.lines, text[i].text.split(/\n/)],
+          timeStamp: [...this.state.timeStamp, data.chunks[i].date],
+        })
+      }
+    }
 
-      //Multiple by 60000 to convert offset to milliseconds
-      var offset   = timer.getTimezoneOffset() * 60000
-      var timezone = timeMil - offset
-      var newDate  = new Date(timezone)
+    var lineItem = this.state.lines;
 
-      var div  = document.createElement('div');
-      var time = document.createElement('div');
+    for(var j = this.state.index; j < lineItem.length; j++){
+      for(var k = 0; k < lineItem[j].length; k++){
 
-      div.className  = 'line';
-      time.className = 'time';
+        var div  = document.createElement('div');
+        var time = document.createElement('a');
 
-      div.innerHTML  = ansi_up.ansi_to_html(data.chunks[i].text)
-      time.innerHTML = moment(newDate).parseZone().format("h:mm:ss a")
+        div.className  = 'line';
+        time.className = 'time';
+        div.id         = 'L' + id++;
 
-      frag.appendChild(time)
-      frag.appendChild(div)
-    }
+        let {environment, number} = this.state.task;
+        let {name}                = this.state.task.app;
+
+        time.href      = `/deploys/${name}/${environment}/${number}#${div.id}`;
+        time.id        = 'L' + timeid++;
 
+        time.addEventListener('click', (e) => {
```

Comment:
This will create a new event listener on every line/`time` element. An alternative is to have an event listener on the container element.